Add Max Memory limit section for trace-agent (#13805)
DataDog/documentation: content/en/tracing/troubleshooting/agent_rate_limits.md

```

To increase the APM connection limit for the Agent, configure the `connection_limit` attribute within the Agent's configuration file (underneath the `apm_config:` section). For containerized deployments (for example, Docker or Kubernetes), use the `DD_APM_CONNECTION_LIMIT` environment variable.

## Max memory limit

> If you encounter the following error message in your Agent logs, the max memory usage of Agent has been exceeded by 150%:

Reviewer: ```suggestion
If you encounter the following error message in your Agent logs, it means the Agent has exceeded the max memory usage by 150%:
```

```
CRITICAL | (pkg/trace/api/api.go:703 in watchdog) | Killing process. Memory threshold exceeded: 8238.08M / 715.26M
CRITICAL | (pkg/trace/osutil/file.go:39 in Exitf) | OOM
```

> To increase the max memory limit for the Agent, configure the `max_memory` attribute within the Agent's configuration file (underneath the `apm_config:` section). For containerized deployments (for example, Docker or Kubernetes), use the `DD_APM_MAX_MEMORY` environment variable.

Reviewer: ```suggestion
To increase the max memory limit for the Agent, configure the `max_memory` attribute in the `apm_config` section of the Agent's configuration file. For containerized deployments (for example, Docker or Kubernetes), use the `DD_APM_MAX_MEMORY` environment variable.
```
